### DISCUSSION TITLE
Carla on Raspberry Pi Small Touch Screen

### DISCUSSION THREAD

**Discussion by RK11111111111** [falkTX/Carla#1871] [Ideas]
I have a project.  The ultimate goal is to have a device that boots directly to a synth, so that kids can use it.  No poking around on the desktop.  Turn it on, wait play, then play the synth with a midi keyboard.  The OS I am using it Raspberry Pi based.  I have built similar devices based on Raspberry Pis before.  A headless minidexed.  Which doesn't use an OS at all and Mixxx a DJ controller software.

I have built a Raspberry pi, however when I launch Carla, it is completely off the touchscreen.  I can use mouse and keyboard to navigate, but eventually I don't want to use these.  EG right click won't work.

I have gotten Surge to work with Carla! The window is just a little large, but not bad.  Any ideas on how to get the device to load jackd, carla, and start with surge open?

I was thinking about maybe forking the Mixxx, repo as it appears to have done something similar booting a Raspberry pi desktop and setting up all the audio configurations, then loading Mixxx.

**Comment by PetGecko** [comment]
You might consider running it headless (no screen).  If your controller keyboard can send program changes messages, and if you use the LV2 plugin for surge, then you can change voices by sending program changes.  And if you want to use multiple plugins you can switch among them by putting them on different MIDI channels (this may require use of the "midi channel filter" plugin, but it's easy to use).  I've set up a Raspberry Pi 400 to work exactly this way:
- MIDI channel 1 is the upper manual of the setBfree organ
- MIDI channel 2 is the lower manual of the setBfree organ (my controller can send out channels 1 and 2 simultaneously)
- MIDI channel 3 is Pianoteq
- MIDI channel 4 is Surge-XT
Your specifics may differ, but the main point is you don't necessarily need a screen.
